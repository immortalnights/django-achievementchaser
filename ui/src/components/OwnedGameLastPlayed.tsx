import { Typography } from "@mui/material"
import { formatDate, getRelativeTime } from "../utilities"

const OwnedGameLastPlayed = ({ lastPlayed }: PlayerOwnedGame) => (
    <div>
        <Typography variant="subtitle1" textTransform="uppercase">
            Last Played
        </Typography>
        {lastPlayed ? (
            <div title={formatDate(lastPlayed)}>
                {getRelativeTime(lastPlayed)}
            </div>
        ) : (
            <div>Never</div>
        )}
    </div>
)

export default OwnedGameLastPlayed
