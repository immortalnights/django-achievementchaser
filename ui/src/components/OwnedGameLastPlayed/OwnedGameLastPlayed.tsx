import { Typography } from "@mui/material"
import { formatDate, getRelativeTime } from "@/dayjsUtilities"

const OwnedGameLastPlayed = ({ lastPlayed }: Omit<PlayerOwnedGame, "game">) => (
    <div>
        <Typography variant="subtitle1" textTransform="uppercase">
            Last Played
        </Typography>
        {lastPlayed ? (
            <div title={formatDate(lastPlayed)}>
                {getRelativeTime(lastPlayed)}
            </div>
        ) : (
            <div>Never</div>
        )}
    </div>
)

export default OwnedGameLastPlayed
