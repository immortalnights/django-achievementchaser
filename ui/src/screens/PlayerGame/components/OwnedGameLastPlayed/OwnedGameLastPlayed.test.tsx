import renderer from "react-test-renderer"
import { expect, test, vi } from "vitest"
import OwnedGameLastPlayed from "."
import { playedPlayerOwnedGame } from "@test/data"

test("snapshot", () => {
    vi.setSystemTime("2024-02-23T12:00:00Z")

    // eslint-disable-next-line
    const component = renderer.create(
        <OwnedGameLastPlayed {...playedPlayerOwnedGame} />
    )
    expect(component.toJSON()).toMatchSnapshot()

    vi.useRealTimers()
})
