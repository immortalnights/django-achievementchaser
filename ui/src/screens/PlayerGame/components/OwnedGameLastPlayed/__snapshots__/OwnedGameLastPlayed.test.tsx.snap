// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1daqgxn-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:1rem;line-height:1.75;letter-spacing:0.00938em;text-transform:uppercase;}",
      }
    }
    data-emotion="css 1daqgxn-MuiTypography-root"
  />
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 css-1daqgxn-MuiTypography-root"
  >
    Last Played
  </h6>
  <div
    title="Feb 11th, 2024"
  >
    13 days ago
  </div>
</div>
`;
